rclc_parameter/include/rclc_parameter/rclc_parameter.h

```diff
@@ -0,0 +1,372 @@
+
+typedef struct rcl_interfaces__srv__SetParameters_Request SetParameters_Request;
+typedef struct rcl_interfaces__srv__SetParameters_Response SetParameters_Response;
+typedef struct rcl_interfaces__msg__SetParametersResult SetParameters_Result;
+
+typedef struct rcl_interfaces__srv__DescribeParameters_Request DescribeParameters_Request;
+typedef struct rcl_interfaces__srv__DescribeParameters_Response DescribeParameters_Response;
+
+typedef struct rcl_interfaces__srv__ListParameters_Request ListParameters_Request;
+typedef struct rcl_interfaces__srv__ListParameters_Response ListParameters_Response;
+
+typedef struct rcl_interfaces__msg__Parameter Parameter;
+typedef struct rcl_interfaces__msg__ParameterValue ParameterValue;
+typedef struct rcl_interfaces__msg__Parameter__Sequence Parameter__Sequence;
+typedef struct rcl_interfaces__msg__ParameterEvent ParameterEvent;
+
+// Number of RCLC executor handles required for a parameter server
+#define RCLC_PARAMETER_EXECUTOR_HANDLES_NUMBER 5
+
+// On parameter modified callback
+typedef void (* ModifiedParameter_Callback)(Parameter * param);
+
+// Allowed RCLC parameter types
+typedef enum rclc_parameter_type_t
+{
+  RCLC_PARAMETER_NOT_SET = 0,
+  RCLC_PARAMETER_BOOL,
+  RCLC_PARAMETER_INT,
+  RCLC_PARAMETER_DOUBLE
+} rclc_parameter_type_t;
+
+// RCLC parameter server options
+typedef struct rclc_parameter_options_t
+{
+  bool notify_changed_over_dds;
+  size_t max_params;
+} rclc_parameter_options_t;
+
+// Container for RCLC parameter server
+typedef struct rclc_parameter_server_t
+{
+  rcl_service_t get_service;
+  rcl_service_t get_types_service;
+  rcl_service_t set_service;
+  rcl_service_t list_service;
+  rcl_service_t describe_service;
+  rcl_publisher_t event_publisher;
+
+  GetParameters_Request get_request;
+  GetParameters_Response get_response;
+
+  GetParameterTypes_Request get_types_request;
+  GetParameterTypes_Response get_types_response;
+
+  SetParameters_Request set_request;
+  SetParameters_Response set_response;
+
+  ListParameters_Request list_request;
+  ListParameters_Response list_response;
+
+  DescribeParameters_Request describe_request;
+  DescribeParameters_Response describe_response;
+
+  Parameter__Sequence parameter_list;
+
+  ParameterEvent event_list;
+
+  ModifiedParameter_Callback on_modification;
+
+  bool notify_changed_over_dds;
+} rclc_parameter_server_t;
+
+/**
+ *  Initializes a RCLC parameter server in a node with default configuration
+ *
+ *  * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | No
+ *
+ * \param[inout] parameter_server preallocated rclc_parameter_server_t
+ * \param[in] node related node
+ * \return `RCL_RET_OK` if the \p parameter_server was initialized successfully
+ */
+
+rcl_ret_t rclc_parameter_server_init_default(
```

Comment:
For windows, visibility_control.h is necessary and a macro for each function. See this pull request https://github.com/ros2/rclc/pull/62

Reply:
Addressed in aa0943898927b96ab9be543e6a3e2071d89be99c